x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/EqlSpecLoader.java

```diff
@@ -55,6 +55,14 @@ private static String getTrimmedString(TomlTable table, String key) {
             spec.note(getTrimmedString(table, "note"));
             spec.description(getTrimmedString(table, "description"));
 
+            // default these values to false
+            spec.caseSensitive(Boolean.TRUE.equals(table.getBoolean("case_sensitive")));
+            spec.caseInsensitive(Boolean.TRUE.equals(table.getBoolean("case_insensitive")));
+
+            if (spec.caseInsensitive() == false && spec.caseSensitive() == false) {
+                throw new IllegalArgumentException("Test must support case-sensitivity or case-insensitivity for test: " + spec.toString());
+            }
```

Comment:
Instead of forcing each query to specify a case sensitivity, the default should be missing an argument means that parameter is `true`.
That is instead of specifying:
```
case_sensitive = true
case_insensitive = true
```
don't specify anything. If something is just case sensitive, it means it's insensitive and vice-versa and when none is specified, apply both (case irrelevant).
This makes the test declaration easier.

Reply:
This is what I used to have, and simplified writing the tests but the behavior didn't seem to be well liked within EqlSpec.

> Instead of forcing each query to specify a case sensitivity, the default should be missing an argument means that parameter is true.

This was the original behavior
```toml
[[queries]]
# this was an error
case_sensitive = true
case_insensitive = true

[[queries]]
# test will be checked with case sensitive mode only
case_sensitive = true

[[queries]]
# test will be checked with case insensitive mode only
case_insensitive = true

[[queries]]
# test will be checked with both case sensitive and case-insensitive mode
# case_sensitive = null
# case_insensitive = null
```

But if missing an argument "means that parameter is `true`", then you basically are inverting the flags. Then you have to negate the logic to denote which tests are _not_ allowed
```toml
[[queries]] 
# the only way to run a test as case-insensitive only
case_sensitive = false
```

Reply:
I think the most desired behavior is this:
* if both `case_sensitive` and `case_insensitive` are `null`, then set them to both to `true`

Reply:
> I think the most desired behavior is this?
> If both `case_sensitive` and `case_insensitive` are `null`, then set them to both to `true`

Correct. Looking at the individual properties in isolation and setting them to true won't work since they *both* have to be missing.